### PR TITLE
All your base

### LINKED ISSUE #162: Add Practice Exercise: `all-your-base`
Problem specification:
https://github.com/exercism/problem-specifications/tree/main/exercises/all-your-base
    
TODO:
    
- generate the exercise with `./bin/fetch-configlet && ./bin/configlet create --practice-exercise all-your-base`
- check if any of the existing tracks (examples below) implement the exercise and you can reuse the implementation:
    - Go: https://github.com/exercism/go/tree/main/exercises/practice
    - Rust: https://github.com/exercism/rust/tree/main/exercises/practice
    - Python: https://github.com/exercism/python/tree/main/exercises/practice
- use exercise's _canonical-data.json_ to see which test cases to implement:
    https://github.com/exercism/problem-specifications/blob/main/exercises/all-your-base/canonical-data.json

### PR DESCRIPTION
**Pull request by Falilah**
Closes #162

### CODE REVIEW

File: exercises/practice/all-your-base/src/lib.cairo [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/// which is equivalent to 101010 in binary.
+///
+///
+/// Notes:
+///  * The empty array ( "[]" ) is equal to the number 0.
+///  * Never output leading 0 digits, unless the input number is 0, in which the output must be
+///  `[0]`.
+///    However, your function must be able to process input with leading 0 digits.
+///
+pub fn convert(number: Array<u32>, from_base: u32, to_base: u32) -> Result<Array<u32>, Error> {
```
Comment (0xNeshi) [inline exercises/practice/all-your-base/src/lib.cairo:40]: Rename the function to `rebase` to align it with [canonical data](https://github.com/exercism/problem-specifications/blob/main/exercises/all-your-base/canonical-data.json#L18)

File: config.json [hunk 2]
```diff
@@ -803,6 +803,14 @@
...
         "prerequisites": [],
         "difficulty": 5
+      },
+      {
+        "slug": "all-your-base",
+        "name": "All Your Base",
+        "uuid": "102e5a3f-c5bb-4b8a-acab-7bdbacec38fa",
+        "practices": [],
+        "prerequisites": [],
+        "difficulty": 1
```
Comment (0xNeshi) [inline config.json:813]: This is a harder exercise, set it to at least 4

File: exercises/practice/all-your-base/tests/all_your_base.cairo [hunk 3]
```diff
@@ -0,0 +1,224 @@
+use all_your_base::{convert, Error};
+
+#[test]
+fn single_bit_one_to_decimal() {
+    let input_base = 2;
+    let input_digits: Array<u32> = array![1];
+    let output_base = 10;
+
+    let expected_output: Array<u32> = array![1];
```
Comment (0xNeshi) [inline exercises/practice/all-your-base/tests/all_your_base.cairo:9]:
```suggestion
    let expected_output = Result::Ok(array![1]);
```
and later below:
```rust
assert_eq!(result, expected_output);
```
It's cleaner

**Review by 0xNeshi — CHANGES_REQUESTED** [review]

File: exercises/practice/all-your-base/src/lib.cairo [hunk 4]
```diff
@@ -5,38 +5,6 @@ pub enum Error {
...
-///
-///
-/// Notes:
-///  * The empty array ( "[]" ) is equal to the number 0.
-///  * Never output leading 0 digits, unless the input number is 0, in which the output must be
-///  `[0]`.
-///    However, your function must be able to process input with leading 0 digits.
-///
-pub fn convert(number: Array<u32>, from_base: u32, to_base: u32) -> Result<Array<u32>, Error> {
+pub fn rebase(digits: Array<u32>, inputBase: u32, outputBase: u32) -> Result<Array<u32>, Error> {
```
Comment (0xNeshi) [inline exercises/practice/all-your-base/src/lib.cairo:8]:
```suggestion
pub fn rebase(digits: Array<u32>, input_base: u32, output_base: u32) -> Result<Array<u32>, Error> {
```
snake_case is idiomatic

**Review by 0xNeshi — APPROVED** [review]
Good job! Polished some things for you, to speed up the review

**Comment by Falilah** [comment]
> Good job! Polished some things for you, to speed up the review

Thank you

**Pull request merged**